### DISCUSSION TITLE
@ea.Problem.single的作用

### DISCUSSION THREAD

**Discussion by Alvin-prc** [geatpy-dev/geatpy#337] [Q&A]
我能理解，这个装饰器让目标函数一次只能传入一个样本，但是我不明白为什么要这样做，既然有没有这个装饰器结果都一样，为什么要写一个这个东西，请问为什么？

**Comment by daytime001** [comment]
不使用这个装饰器，每次传入的就是种群大小 组解啊，同时对整个种群进行计算，这样就必须使用向量化操作或者for循环。使用这个装饰器之后一次只能传入一组解，更容易理解，更符合一般算法的应用逻辑
